### PR TITLE
Tidy Up and Create Some Utils

### PR DESCRIPTION
**Pull request by muhammadridho**
Halo, Halo.

I'm really interesting about your project, so hopefully my contribution can help. this is some list what happened

1. Restructured Code
2. Create Decoder into model
3. Create HOC for Http Request

### CODE REVIEW

**Comment by mazipan** [comment]
CI nya failed mas @muhammadridho, bisa bantuin benerin sekalian gak ya.
Abis itu baru di merge.

Thanks for your help 👍

File: src/common/Model.re [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  imdb: option(string),
+  image: option(string),
+  streaming: option(string)
+};
+
+
+let read_film = (json) => {
+    let id = Json.Decode.(json |> field("id", string));
+    let staticData = Utils.Data.getAdditionalData(id);
+    Js.log(json);
```
Comment (mazipan) [inline src/common/Model.re]: Remove log ya mas

File: src/common/Constants.re [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
   image: string,
   streaming: string,
-  title: string,
+  title:string,
 };
 
-let filmConstantData: list(filmConstantModel) = [
+module Url = {
+    let base = "https://ghibliapi.herokuapp.com/films";
+    let get_detail_from_id = (id) => {j|$base/$id|j};
```
Comment (mazipan) [inline src/common/Constants.re]: Saya gak tau Reason itu memang prefer snake case atau camelCase ya mas?

File: package.json [hunk 3]
```diff
@@ -6,11 +6,11 @@
...
   "scripts": {
     "test": "NODE_ENV=production npm run build",
-    "start": "npm run dev | npm run bsb-watch",
+    "start": "npm run dev | npm run watch:re",
     "dev": "NODE_ENV=development webpack-dev-server",
-    "bsb-watch": "bsb -make-world -w",
+    "watch:re": "bsb -make-world -w",
     "build": "npm run clean:dist && npm run build:bsb && npm run build:webpack",
-    "build:bsb": "bsb -make-world",
+    "build:re": "bsb -make-world",
```
Comment (mazipan) [inline package.json]: Ini yang bikin CI nya ngambek, sekalian benerin file .travis.yml mas

File: src/common/Model.re (hunk 1 above)
Comment (muhammadridho) [reply src/common/Model.re]: oh ya wkwwk

File: package.json (hunk 3 above)
Comment (muhammadridho) [reply package.json]: ok

**Comment by muhammadridho** [comment]
Updated @mazipan

**Comment by mazipan** [comment]
@all-contributors please add @muhammadridho for code

**Pull request merged**